lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -435,6 +441,17 @@ sub _compose_job_overview_search_args ($c) {
         my @search_terms = (@group_id_search, @group_name_search);
         @groups = $schema->resultset('JobGroups')->search(\@search_terms)->all;
     }
+
+    # use globs to filter job groups, first include all groups that match "group_glob" values, and then exclude those
+    # that also match "not_group_glob" values
+    elsif (@$group_glob || @$not_group_glob) {
+        my @inclusive = map { qr/$_/i } map { glob_to_regex_string($_) } @$group_glob;
```

Comment:
Maybe avoid the double `map` calls (which in Perl likely result in an intermediate array, right?):
```suggestion
        my @inclusive = map { my $regex = glob_to_regex_string($_); qr/$regex/i } @$group_glob;
```

Reply:
Too cheap to worry about resource usage, the intermediate list costs nothing really. Most readable solution wins here imo.